keras_nlp/models/bart/bart_seq_2_seq_preprocessor.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""BART Seq2Seq preprocessor layer."""
+
+from absl import logging
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.models.bart.bart_preprocessor import BartPreprocessor
+from keras_nlp.utils.keras_utils import pack_x_y_sample_weight
+
+
+@keras_nlp_export("keras_nlp.models.BartSeq2SeqPreprocessor")
+class BartSeq2SeqPreprocessor(BartPreprocessor):
+    """BART Seq2Seq preprocessor.
+
+    This layer is used as preprocessor for seq2seq tasks using the BART model.
+    This class subclasses `keras_nlp.models.BartPreprocessor` and keeps most of
+    its functionality. It has two changes from the superclass:
+
+     - Sets the `y` (label) and `sample_weights` fields by shifting the
+       decoder input sequence one step towards the left. Both these fields are
+       inferred internally, and any passed values will be ignored.
+     - Drops the last token from the decoder input sequence as it does not have
+       a successor.
+
+    Args:
+        tokenizer: A `keras_nlp.models.BartTokenizer` instance.
+        sequence_length: The length of the packed inputs.
```

Comment:
Probably worth mentioning that this is the length for both encoder and decoder sequences (for now).